docs: add documentation for all secure acl introduction work

rboyer opened the pull request:
When fixing some style consistency issues I carried through and tried to update similar issues in pre-existing docs (like inconsistent use of `hcl/sh` codeblock formatting).

kaitlincarter-hc commented on website/source/layouts/api.erb:
```diff
@@ -22,6 +22,15 @@
           <li<%= sidebar_current("api-acl-policies") %>>
             <a href="/api/acl/policies.html">Policies</a>
           </li>
+          <li<%= sidebar_current("api-acl-roles") %>>
```
I think we should spend a few minutes talking about the order of this list. The whole thing could probably be updated.

kaitlincarter-hc commented on website/source/api/acl/role-binding-rules.html.md:
```diff
@@ -0,0 +1,371 @@
+
+# ACL Role Binding Rule HTTP API
+
+The `/acl/rolebindingrule` endpoints [create](#create-a-role-binding-rule),
```
Is it too late to rename this endpoint? The `rolebindingrule` endpoints it a little long and hard to read.

rboyer replied:
What name would you suggest?

johncowen replied:
I think we should bikeshed this a little more in Slack maybe, I also brought this up. Agree that `rolebindingrule` might be the best name for it, but if we can get a shorter one it might be better. If we can't come up with anything else between us then fairynuff 🤷‍♂️

kaitlincarter-hc replied:
Maybe we can discuss on the community call Friday? My initial thoughts are:
- `binding` since role is already an endpoint. Also this seems like the only binding that we will have with the ACLs (we should think into the future). 
- `bindingrules` or `bind-rules` or `bindrules` this is a little more specific, but becomes harder to use. 
- `rules`, but this is in danger of being overloaded.

kaitlincarter-hc commented on website/source/api/acl/identity-providers.html.md:
```diff
@@ -0,0 +1,328 @@
+    "Type": "kubernetes",
+    "KubernetesHost": "https://192.0.2.42:8443",
+    "KubernetesCACert": "-----BEGIN CERTIFICATE-----\nMIIC5zCCAc+gAwIBAgIBATANBgkqhkiG9w0BAQsFADAVMRMwEQYDVQQDEwptaW5p\na3ViZUNBMB4XDTE5MDEwODE4NTAwNFoXDTI5MDEwNjE4NTAwNFowFTETMBEGA1UE\nAxMKbWluaWt1YmVDQTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBALna\nHbH7BirRjzbExPgGdsikbPUtwmUqyGvJ5SGGho0RkkL/HIL5D154CD8CvAMEbSMt\n20FBqbI5h5JeJ9gYUdsPLwnMVI2r2yvJuzuPxtxiyf63DU7XxXi3v6QP6J0qRhDr\nlEhp5K2Xrd7qIxA ...
+    "KubernetesServiceAccountJWT": "eyJhbGciOiJSUzI1NiIsImtpZCI6IiJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJkZWZhdWx0Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZWNyZXQubmFtZSI6ImNvbnN1bC1pZHAtdG9rZW4tcmV2aWV3LWFjY291bnQtdG9rZW4tbHBteHYiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlcnZpY2UtYWNjb3VudC5uYW1lIjoiY29uc3VsLWlkcC10b2tlbi1yZX ...
```
I think we should truncate all these tokens and certificates. If necessary we can call that out at the beginning of the doc.

rboyer replied:
I'll just do it how I noticed the vault docs do it: https://www.vaultproject.io/api/auth/kubernetes/index.html

kaitlincarter-hc commented on website/source/api/acl/roles.html.md:
```diff
@@ -0,0 +1,462 @@
+
+The `/acl/role` endpoints [create](#create-a-role), [read](#read-a-role),
+[update](#update-a-role), [list](#list-roles) and [delete](#delete-a-role)  ACL roles in Consul.
+For more information about ACLs, please see the [ACL Guide](/docs/guides/acl.html).
```
This link will need to be updated, depending on where we document `roles`. If we decide to add it to the guides then we will need to link out to Learn. However, I thought we wanted to wait on adding roles to the guides since its a bit complicated for users who aren't using IdPs.

rboyer replied:
That stub I left in place when cloning the Policy skeleton for now, but I agree that any links outside of `/api` are subject to change.

kaitlincarter-hc commented on website/source/api/acl/roles.html.md:
```diff
@@ -0,0 +1,462 @@
+[blocking queries](/api/index.html#blocking-queries),
+[consistency modes](/api/index.html#consistency-modes),
+[agent caching](/api/index.html#agent-caching), and
+[required ACLs](/api/index.html#acls).
```
*Warning* This the /api/index.html page might be updated with the filtering PR. It will affect the location of these sections (expect maybe ACLS)

kaitlincarter-hc commented on website/source/api/acl/roles.html.md:
```diff
@@ -0,0 +1,462 @@
+### Parameters
+
+- `Name` `(string: <required>)` - Specifies a name for the ACL role. The name
+   can only contain alphanumeric characters as well as `-` and `_` and must be
```
```suggestion
   can only contain alphanumeric characters, dashes `-` , underscores `_`,  and must be
```
If someone is using a screen reader it probably won't recognize just `_` or `-`

rboyer replied:
That's actually text I cloned from the policies page (so it might be worth updating the same thing over there unrelated to my PR). I left a TODO because I need to update the validation text as it is more restrictive than policy names.

kaitlincarter-hc replied:
I think in this case its ok to be inconsistent for now, until we make bigger updates. 

We should maybe use an accessibility tool to catch all these things on the site and then make one big update.....

kaitlincarter-hc commented on website/source/api/acl/roles.html.md:
````diff
@@ -0,0 +1,462 @@
+
+### Sample Request
+
+```text
````
```suggestion
```sh
```

kaitlincarter-hc commented on website/source/api/acl/roles.html.md:
```diff
@@ -0,0 +1,462 @@
+   internally resolved to the policy ID. With linking roles internally by IDs,
+   Consul enables policy renaming without breaking tokens.
+
+- `ServiceIdentities` `(array<ServiceIdentity>)` - The list of service
```
I think I still struggle a little with this concept. Maybe we can link out to the location where we define `service identity`

kaitlincarter-hc commented on website/source/api/acl/roles.html.md:
````diff
@@ -0,0 +1,462 @@
+
+### Sample Request
+
+```text
````
```suggestion
```sh
```

kaitlincarter-hc commented on website/source/api/acl/roles.html.md:
````diff
@@ -0,0 +1,462 @@
+
+### Sample Request
+
+```text
````
```suggestion
```sh
```

kaitlincarter-hc commented on website/source/api/acl/roles.html.md:
````diff
@@ -0,0 +1,462 @@
+
+### Sample Request
+
+```text
````
```suggestion
```sh
```

kaitlincarter-hc commented on website/source/api/acl/roles.html.md:
````diff
@@ -0,0 +1,462 @@
+
+### Sample Request
+
+```text
````
```suggestion
```sh
```

kaitlincarter-hc commented on website/source/api/acl/roles.html.md:
````diff
@@ -0,0 +1,462 @@
+
+## Sample Request
+
+```text
````
```suggestion
```sh
```

kaitlincarter-hc commented on website/source/api/acl/tokens.html.md:
```diff
@@ -252,10 +287,41 @@ The table below shows this endpoint's support for
+
+~> TODO:where is the best place to mention RoleLink.BoundName?
+
+- `ServiceIdentities` `(array<ServiceIdentity>)` - The list of service
```
Again, maybe we could link out here to the `service identity definition`

kaitlincarter-hc commented on website/source/api/acl/tokens.html.md:
````diff
@@ -414,7 +480,7 @@ The table below shows this endpoint's support for
 ### Sample Request
 
 ```text
````
```suggestion
```sh
```

kaitlincarter-hc commented on website/source/docs/acl/acl-system.html.md:
```diff
@@ -27,8 +27,27 @@ At the highest level, there are two major components to the ACL system:
 
- ACL tokens and policies are managed by Consul operators via Consul's
-[ACL API](/api/acl/acl.html), [ACL CLI](/docs/commands/acl.html), or systems like
+For many scenarios policies and tokens are sufficient, but more advanced setups
```
I like the way you've divided these concepts.

kaitlincarter-hc commented on website/source/docs/acl/acl-system.html.md:
````diff
@@ -46,6 +66,42 @@ An ACL policy is a named set of rules and is composed of the following elements:
+At authorization time a service identity acts like an additional policy were
+attached to the token with the following contents:
+
+```
````
```suggestion
```json
```

Will JSON get angry with the `//`?

kaitlincarter-hc commented on website/source/api/acl/role-binding-rules.html.md:
```diff
@@ -0,0 +1,371 @@
+| `PUT`  | `/acl/rolebindingrule`       | `application/json`         |
+
+The table below shows this endpoint's support for
+[blocking queries](/api/index.html#blocking-queries),
```
Same note here, for the entire doc, these file locations may change.

kaitlincarter-hc commented on website/source/api/acl/role-binding-rules.html.md:
```diff
@@ -0,0 +1,371 @@
+  role must already exist for this role binding rule to apply. This is described
+  in more detail in the [Identity Provider Guide](/docs/guides/acl-identity-providers.html).
+
+- `Matches` `(array<Match>)` - The list of match selectors. Individual matches
```
**Warning** Possible user confusion.
Are we overloading terms here? Does a match selector here mean the same as with filtering? If not should we use different terms? @mkeeler

kaitlincarter-hc replied:
Same goes for `selector`

rboyer replied:
If @mkeeler thinks his new stuff is general purpose enough I could take a stab at rebasing the role binding rule syntax to use his new selector syntax instead.

kaitlincarter-hc commented on website/source/api/acl/role-binding-rules.html.md:
````diff
@@ -0,0 +1,371 @@
+
+### Sample Request
+
+```text
````
```suggestion
```sh
```

kaitlincarter-hc commented on website/source/api/acl/role-binding-rules.html.md:
````diff
@@ -0,0 +1,371 @@
+
+### Sample Request
+
+```text
````
```suggestion
```sh
```

kaitlincarter-hc commented on website/source/api/acl/role-binding-rules.html.md:
````diff
@@ -0,0 +1,371 @@
+
+### Sample Request
+
+```text
````
```suggestion
```sh
```

kaitlincarter-hc commented on website/source/api/acl/role-binding-rules.html.md:
````diff
@@ -0,0 +1,371 @@
+
+### Sample Request
+
+```text
````
```suggestion
```sh
```

kaitlincarter-hc commented on website/source/api/acl/role-binding-rules.html.md:
````diff
@@ -0,0 +1,371 @@
+
+## Sample Request
+
+```text
````
```suggestion
```sh
```

kaitlincarter-hc commented on website/source/api/acl/identity-providers.html.md:
````diff
@@ -0,0 +1,328 @@
+
+### Sample Request
+
+```text
````
```suggestion
```sh
```

kaitlincarter-hc commented on website/source/api/acl/identity-providers.html.md:
```diff
@@ -0,0 +1,328 @@
+  The /acl/idp endpoints manage Consul's ACL Identity Providers.
+---
+
+-> **1.5.0+:**  The APIs are available in Consul versions 1.5.0 and later. The documentation for the legacy ACL API is [here](/api/acl/legacy.html)
```
I'm not sure we want to refer people to legacy docs, since they could be using the new system just not the latest. This seems like a group discussion.

kaitlincarter-hc replied:
This will apply to all three new API docs

rboyer replied:
Maybe because these concepts are completely new and you are far less likely to end up on these pages by accident I could omit the second sentence in these warnings?

kaitlincarter-hc replied:
I think its ok to omit the second sentence. I would hope by this point in the documentation the users are aware there are two different ACL systems and are being careful. We can revisit if users start having issues here.

kaitlincarter-hc commented on website/source/api/acl/identity-providers.html.md:
```diff
@@ -0,0 +1,328 @@
+### Parameters
+
+- `Name` `(string: <required>)` - Specifies a name for the ACL identity
+  provider. The name can only contain alphanumeric characters as well as `-`
```
```suggestion
  provider. The name can only contain alphanumeric characters, dashes `-`,
```

kaitlincarter-hc commented on website/source/api/acl/identity-providers.html.md:
```diff
@@ -0,0 +1,328 @@
+
+- `Name` `(string: <required>)` - Specifies a name for the ACL identity
+  provider. The name can only contain alphanumeric characters as well as `-`
+  and `_` and must be unique. This field is immutable.
```
```suggestion
 underscores `_`, and must be unique. This field is immutable.
```

kaitlincarter-hc commented on website/source/api/acl/identity-providers.html.md:
````diff
@@ -0,0 +1,328 @@
+
+### Sample Request
+
+```text
````
```suggestion
```sh
```

kaitlincarter-hc commented on website/source/api/acl/identity-providers.html.md:
````diff
@@ -0,0 +1,328 @@
+
+### Sample Request
+
+```text
````
```suggestion
```sh
```

kaitlincarter-hc commented on website/source/api/acl/identity-providers.html.md:
````diff
@@ -0,0 +1,328 @@
+
+### Sample Request
+
+```text
````
```suggestion
```sh
```

kaitlincarter-hc commented on website/source/api/acl/identity-providers.html.md:
````diff
@@ -0,0 +1,328 @@
+
+## Sample Request
+
+```text
````
```suggestion
```sh
```

kaitlincarter-hc commented:
Overall the API docs look good and consistent with the others! I left some comments inline, there are a few places that need further discussion.

johncowen commented:
Hey 👋 Looking for the best place to make a note of this.

Should we add the new properties for tokens in here:

https://github.com/hashicorp/consul/blob/f-acl-ux-docs/website/source/api/acl/tokens.html.md#sample-response-6

I _think_ this can also now have `Roles` and `ServiceIdentities`? That could be `null` or an `Array`?

kaitlincarter-hc commented on website/source/docs/acl/acl-identity-providers.html.md:
```diff
@@ -0,0 +1,163 @@
+to allow applications to authenticate to Consul using existing credentials and
+receive ACL Tokens usable within the local datacenter in exchange.
+
+The only supported type of identity provider in Consul 1.5 is `kubernetes`
```
```suggestion
The only supported type of identity provider in Consul 1.5 is Kubernetes
```

kaitlincarter-hc commented on website/source/docs/acl/acl-identity-providers.html.md:
```diff
@@ -0,0 +1,163 @@
+receive ACL Tokens usable within the local datacenter in exchange.
+
+The only supported type of identity provider in Consul 1.5 is `kubernetes`
+but it is expected that more will come later.
```
I tend to think we should leave out any future plans/roadmap language in the docs and guides.

kaitlincarter-hc commented on website/source/docs/acl/acl-identity-providers.html.md:
```diff
@@ -0,0 +1,163 @@
+  details of configuration are type dependent and described below.
+
+* **Authorization** - One or more Role Binding Rules must be configured
+  definiting how to translate trusted identity attributes into privileges
```
```suggestion
  defining how to translate trusted identity attributes into privileges
```

kaitlincarter-hc commented on website/source/docs/acl/acl-identity-providers.html.md:
```diff
@@ -0,0 +1,163 @@
+  `consul acl rolebindingrule` subcommands or the corresponding [API
+  endpoints](/api/acl/role-binding-rules.html).
+
+## Login Process
```
I like that you are mostly clear about who is taking the action in these steps, but some places it's still a little fuzzy. Mainly, who sets up RBCs. It would also be nice to have an opening paragraph.

kaitlincarter-hc commented on website/source/docs/acl/acl-identity-providers.html.md:
```diff
@@ -0,0 +1,163 @@
+   endpoint](/api/acl/acl.html#logout-from-identity-provider) to destroy their
+   token when it is no longer required.
+
+## Kubernetes Identity Provider
```
I'm feeling this section should probably be on it's own page probably still under ACLs category. Originally, I thought it should live with the other Kubernetes docs but then where would Vault go?

kaitlincarter-hc commented on website/source/docs/acl/acl-identity-providers.html.md:
```diff
@@ -0,0 +1,163 @@
+
+
+
+### Role Binding Rules
```
The rest of this doc very much feels like a WIP. I would like to see RBRs and Login/Roles under the same section (not in the Kubernetes section) closer to the top. I think each term should have preferably one, max two, paragraphs of information.

kaitlincarter-hc commented on website/source/api/acl/acl.html.md:
```diff
@@ -263,3 +263,112 @@ agent_prefix "" {
+## Login to Auth Method
+
+This endpoint was added in Consul 1.5.0 and is used to exchange an [auth
+method](/docs/acl/acl-auth-methods.html) bearer tokens for a newly-created
```
```suggestion
method](/docs/acl/acl-auth-methods.html) bearer token for a newly-created
```
Assuming you didn't mean this to be plural. If its meant to plural, remove the "an"  from the previous line.

kaitlincarter-hc commented on website/source/api/acl/acl.html.md:
```diff
@@ -263,3 +263,112 @@ agent_prefix "" {
+| `POST` | `/acl/login`                 | `application/json`         |
+
+The table below shows this endpoint's support for
+[blocking queries](/api/index.html#blocking-queries),
```
```suggestion
[blocking queries](/api/features/blocking.html),
```

rboyer replied:
I think this change should be done independently, as master still lists a whole bunch of references to the anchor:
```
consul (master)$ git grep "/api/index.html#blocking-queries" website/source | wc -l
128
```

kaitlincarter-hc commented on website/source/api/acl/acl.html.md:
```diff
@@ -263,3 +263,112 @@ agent_prefix "" {
+
+The table below shows this endpoint's support for
+[blocking queries](/api/index.html#blocking-queries),
+[consistency modes](/api/index.html#consistency-modes),
```
```suggestion
[consistency modes](/api/features/consistency.html),
```

kaitlincarter-hc commented on website/source/api/acl/acl.html.md:
```diff
@@ -263,3 +263,112 @@ agent_prefix "" {
+The table below shows this endpoint's support for
+[blocking queries](/api/index.html#blocking-queries),
+[consistency modes](/api/index.html#consistency-modes),
+[agent caching](/api/index.html#agent-caching), and
```
```suggestion
[agent caching](/api/features/caching.html), and
```

kaitlincarter-hc commented on website/source/api/acl/acl.html.md:
```diff
@@ -263,3 +263,112 @@ agent_prefix "" {
+[blocking queries](/api/index.html#blocking-queries),
+[consistency modes](/api/index.html#consistency-modes),
+[agent caching](/api/index.html#agent-caching), and
+[required ACLs](/api/index.html#acls).
```
To-do: Update this link.

kaitlincarter-hc commented on website/source/api/acl/acl.html.md:
````diff
@@ -263,3 +263,112 @@ agent_prefix "" {
 }
 ```
+
+## Login to Auth Method
````
Do we always want to abbreviate to "Auth Method"? How do others products talk about this, do they say auth method or authentication method? My biggest concern is consistency, but I'm just starting my review so maybe we are consistent.

rboyer replied:
The name of the thing is "auth method" so I was sticking with style used on https://www.vaultproject.io/docs/auth/

kaitlincarter-hc commented on website/source/api/acl/acl.html.md:
```diff
@@ -263,3 +263,112 @@ agent_prefix "" {
+- `AuthMethod` `(string: <required>)` - The name of the auth method to use for login.
+
+- `BearerToken` `(string: <required>)` - The bearer token to present to the
+  auth method during login for authentication purposes. For the Kubernetes Auth
```
By capitalizing "Kubernetes Auth Method" we are putting emphasis here, is that intentional? Typically, only products should be capitalized.

rboyer replied:
fixed

kaitlincarter-hc commented on website/source/api/acl/acl.html.md:
```diff
@@ -263,3 +263,112 @@ agent_prefix "" {
+| `POST` | `/acl/logout`                | `application/json`         |
+
+The table below shows this endpoint's support for
+[blocking queries](/api/index.html#blocking-queries),
```
```suggestion
[blocking queries](/api/features/blocking.html),
```

kaitlincarter-hc commented on website/source/api/acl/acl.html.md:
```diff
@@ -263,3 +263,112 @@ agent_prefix "" {
+
+The table below shows this endpoint's support for
+[blocking queries](/api/index.html#blocking-queries),
+[consistency modes](/api/index.html#consistency-modes),
```
```suggestion
[consistency modes](/api/features/consistency.html),
```

kaitlincarter-hc commented on website/source/api/acl/acl.html.md:
```diff
@@ -263,3 +263,112 @@ agent_prefix "" {
+The table below shows this endpoint's support for
+[blocking queries](/api/index.html#blocking-queries),
+[consistency modes](/api/index.html#consistency-modes),
+[agent caching](/api/index.html#agent-caching), and
```
```suggestion
[agent caching](/api/features/caching.html), and
```

kaitlincarter-hc commented on website/source/api/acl/acl.html.md:
```diff
@@ -263,3 +263,112 @@ agent_prefix "" {
+[blocking queries](/api/index.html#blocking-queries),
+[consistency modes](/api/index.html#consistency-modes),
+[agent caching](/api/index.html#agent-caching), and
+[required ACLs](/api/index.html#acls).
```
to-do: update link

kaitlincarter-hc commented on website/source/api/acl/auth-methods.html.md:
```diff
@@ -0,0 +1,307 @@
+  The /acl/auth-method endpoints manage Consul's ACL Auth Methods.
+---
+
+-> **1.5.0+:**  The APIs are available in Consul versions 1.5.0 and later.
```
```suggestion
-> **1.5.0+:**  The auth method APIs are available in Consul versions 1.5.0 and newer.
```

kaitlincarter-hc commented on website/source/api/acl/auth-methods.html.md:
```diff
@@ -0,0 +1,307 @@
+[read](#read-an-auth-method), [update](#update-an-auth-method),
+[list](#list-auth-methods) and [delete](#delete-an-auth-method)
+ACL auth methods in Consul.  For more information about ACLs, please see
+the [ACL Guide](/docs/guides/acl.html).
```
```suggestion
the [ACL Guide](https://learn.hashicorp.com/consul/advanced/day-1-operations/production-acls).
```

kaitlincarter-hc replied:
```suggestion
the [ACL Guide](https://learn.hashicorp.com/consul/advanced/day-1-operations/production-acls).
```

kaitlincarter-hc commented on website/source/api/acl/auth-methods.html.md:
```diff
@@ -0,0 +1,307 @@
+The `/acl/auth-method` endpoints [create](#create-an-auth-method),
+[read](#read-an-auth-method), [update](#update-an-auth-method),
+[list](#list-auth-methods) and [delete](#delete-an-auth-method)
+ACL auth methods in Consul.  For more information about ACLs, please see
```
```suggestion
ACL auth methods in Consul.  For more information on how to setup ACLs, please see
```

kaitlincarter-hc commented on website/source/api/acl/auth-methods.html.md:
```diff
@@ -0,0 +1,307 @@
+| `PUT`  | `/acl/auth-method`           | `application/json`         |
+
+The table below shows this endpoint's support for
+[blocking queries](/api/index.html#blocking-queries),
```
```suggestion
[blocking queries](/api/features/blocking.html),
```

kaitlincarter-hc commented on website/source/api/acl/auth-methods.html.md:
```diff
@@ -0,0 +1,307 @@
+
+The table below shows this endpoint's support for
+[blocking queries](/api/index.html#blocking-queries),
+[consistency modes](/api/index.html#consistency-modes),
```
```suggestion
[consistency modes](/api/features/consistency.html),
```

kaitlincarter-hc commented on website/source/api/acl/auth-methods.html.md:
```diff
@@ -0,0 +1,307 @@
+The table below shows this endpoint's support for
+[blocking queries](/api/index.html#blocking-queries),
+[consistency modes](/api/index.html#consistency-modes),
+[agent caching](/api/index.html#agent-caching), and
```
```suggestion
[agent caching](/api/features/caching.html), and
```

kaitlincarter-hc commented on website/source/api/acl/auth-methods.html.md:
```diff
@@ -0,0 +1,307 @@
+[blocking queries](/api/index.html#blocking-queries),
+[consistency modes](/api/index.html#consistency-modes),
+[agent caching](/api/index.html#agent-caching), and
+[required ACLs](/api/index.html#acls).
```
needs to be updated

rboyer replied:
Where should the `#acls` anchor tag point now?

kaitlincarter-hc commented on website/source/api/acl/auth-methods.html.md:
```diff
@@ -0,0 +1,307 @@
+### Parameters
+
+- `Name` `(string: <required>)` - Specifies a name for the ACL auth method. The
+  name can only contain alphanumeric characters as well as `-` and `_` and must
```
```suggestion
  name can contain alphanumeric characters, dashes `-`, and  underscores `_`.
```

rboyer replied:
I think I said this earlier, but do you want me to just find all occurrences of this phrasing across all of the ACL docs and fix them on this branch?

kaitlincarter-hc commented on website/source/api/acl/auth-methods.html.md:
```diff
@@ -0,0 +1,307 @@
+
+- `Name` `(string: <required>)` - Specifies a name for the ACL auth method. The
+  name can only contain alphanumeric characters as well as `-` and `_` and must
+  be unique. This field is immutable.
```
```suggestion
 This field is immutable and must be unique.
```

kaitlincarter-hc commented:
Overall, the new documentation looks great! For the HTTP API docs I had to update all the broken links. I also removed a lot of capitalization. I still need to finish the Kubernetes auth method doc though. 

Most of my suggestions are minor, I do not see any blockers. The one area of improvement is around service identities. I'm not sure I got a clear picture of what it is or what it does.

rboyer commented on website/source/api/acl/binding-rules.html.md:
```diff
@@ -0,0 +1,370 @@
+  The /acl/binding-rule endpoints manage Consul's ACL Binding Rules.
+---
+
+-> **1.5.0+:** The APIs are available in Consul versions 1.5.0 and later.
```
```suggestion
-> **1.5.0+:** The binding rule APIs are available in Consul versions 1.5.0 and later.
```

kaitlincarter-hc commented on website/source/docs/acl/acl-auth-methods.html.md:
```diff
@@ -0,0 +1,127 @@
+page_title: "ACL Auth Methods"
+sidebar_current: "docs-acl-auth-methods"
+description: |-
+  An Auth Method is a component in Consul that performs authentication against a trusted external party to authorize the creation of an appropriately scoped ACL Token usable within the local datacenter.
```
```suggestion
  An auth method is a component in Consul that performs authentication against a trusted external party to authorize the creation of an appropriately scoped ACL Token usable within the local datacenter.
```
"Auth Method" or "auth method" we should be consistent.

kaitlincarter-hc commented on website/source/docs/acl/acl-auth-methods.html.md:
```diff
@@ -0,0 +1,127 @@
+# ACL Auth Methods
+
+An Auth Method is a component in Consul that performs authentication against a
+trusted external party to authorize the creation of an appropriately scoped ACL
```
What does appropriately scoped mean? Correct permissions?

kaitlincarter-hc commented on website/source/docs/acl/acl-auth-methods.html.md:
```diff
@@ -0,0 +1,127 @@
+
+An Auth Method is a component in Consul that performs authentication against a
+trusted external party to authorize the creation of an appropriately scoped ACL
+Token usable within the local datacenter.
```
```suggestion
token usable within the local datacenter.
```

kaitlincarter-hc commented on website/source/docs/acl/acl-auth-methods.html.md:
```diff
@@ -0,0 +1,127 @@
+trusted external party to authorize the creation of an appropriately scoped ACL
+Token usable within the local datacenter.
+
+The only supported type of auth method in Consul 1.5 is
```
```suggestion
The only supported type of auth method in Consul 1.5.0 is
```

kaitlincarter-hc commented on website/source/docs/acl/acl-auth-methods.html.md:
```diff
@@ -0,0 +1,127 @@
+
+## Overview
+
+Without auth methods a trusted operator needs to be critically involved in the
```
```suggestion
Without an auth method a trusted operator is critically involved in the
```

kaitlincarter-hc commented on website/source/docs/acl/acl-auth-methods.html.md:
```diff
@@ -0,0 +1,127 @@
+## Overview
+
+Without auth methods a trusted operator needs to be critically involved in the
+creation and secure introduction of each ACL Token to every application that
```
```suggestion
creation and secure introduction of each ACL token to every application that
```

kaitlincarter-hc commented on website/source/docs/acl/acl-auth-methods.html.md:
```diff
@@ -0,0 +1,127 @@
+When running in environments such as a public cloud or when supervised by a
+cluster scheduler, applications may already have access to uniquely identifying
+credentials that were delivered securely by the platform. Consul auth method
+integrations allow for these credentials to be used to create ACL Tokens with
```
```suggestion
integrations allow for these credentials to be used to create ACL tokens with
```

kaitlincarter-hc commented on website/source/docs/commands/acl/acl-role.html.md.erb:
```diff
@@ -0,0 +1,315 @@
+of the entire UUID. As long as it is unique it will be resolved to the full
+UUID and used.
+
+## Common Subcommand Options
```
If we show all subcommand options, we should remove "common".

kaitlincarter-hc commented on website/source/docs/commands/acl/acl-role.html.md.erb:
```diff
@@ -0,0 +1,315 @@
+$ consul acl role list -meta
+web-crawler:
+   ID:           57147d87-6bf7-f794-1a6e-7d038c4e4ae9
+   Description:  better description
```
Can you add a better description :) There are multiple places where this could be updated.

kaitlincarter-hc commented on website/source/docs/commands/index.html.md:
```diff
@@ -40,6 +42,8 @@ Available commands are:
     kv             Interact with the key-value store
     leave          Gracefully leaves the Consul cluster and shuts down
     lock           Execute a command holding a lock
+    login          Login to Consul using an Auth Method
```
Should "Auth Method" be capitalized?

rboyer commented on website/source/docs/acl/acl-auth-methods.html.md:
```diff
@@ -0,0 +1,127 @@
+   with the specified auth method and selects only those that match the trusted
+   identity attributes.
+
+6. Bound roles and service identites are computed. If none are computed the
```
kait: Where is 6 and 7 happening? So far steps 1-5 are clear and easy to follow, but I get a little lost here. Does it make sense to have two different lists for this process?

kaitlincarter-hc commented on website/source/docs/commands/index.html.md:
```diff
@@ -40,6 +42,8 @@ Available commands are:
     leave          Gracefully leaves the Consul cluster and shuts down
     lock           Execute a command holding a lock
+    login          Login to Consul using an Auth Method
+    logout         Destroy a Consul Token created with Login
```
```suggestion
    logout         Destroy a Consul token created with Login
```

rboyer commented on website/source/docs/acl/acl-auth-methods.html.md:
```diff
@@ -0,0 +1,127 @@
+   [logout API endpoint](/api/acl/acl.html#logout-from-auth-method) to destroy
+   their token when it is no longer required.
+
+For more details about specific auth methods and how to configure them, consult
```
kait: Can we link it here?

kaitlincarter-hc commented on website/source/docs/commands/login.html.md.erb:
```diff
@@ -0,0 +1,50 @@
+page_title: "Commands: Login"
+sidebar_current: "docs-commands-login"
+description: >
+  The `login` command provides control of both service and node loginenance mode
```
loginenance??

rboyer replied:
```suggestion
  The `login` command will exchange the provided third party credentials with the
requested auth method for a newly minted Consul ACL Token.
```

kaitlincarter-hc commented on website/source/docs/commands/login.html.md.erb:
```diff
@@ -0,0 +1,50 @@
+Command: `consul login`
+
+The `login` command will exchange the provided third party credentials with the
+requested auth method for a newly minted Consul ACL Token. The companion
```
```suggestion
requested auth method for a newly minted Consul ACL token. The companion
```

rboyer replied:
```suggestion
requested auth method for a newly minted Consul ACL token. The companion
```

rboyer commented on website/source/docs/acl/acl-system.html.md:
```diff
@@ -27,16 +27,35 @@ At the highest level, there are two major components to the ACL system:
+For many scenarios policies and tokens are sufficient, but more advanced setups
+may benefit from additional components in the ACL system:
+
+ * **ACL Roles** - Roles allow for grouping a set of policies and service
```
```suggestion
 * **ACL Roles** - Roles allow for the grouping of a set of policies and service
```

rboyer commented on website/source/docs/acl/acl-system.html.md:
```diff
@@ -27,16 +27,35 @@ At the highest level, there are two major components to the ACL system:
+may benefit from additional components in the ACL system:
+
+ * **ACL Roles** - Roles allow for grouping a set of policies and service
+   identities into a reusable higher-level entity that can be linked with many
```
```suggestion
   identities into a reusable higher-level entity that can be applied to many
```

kaitlincarter-hc commented on website/source/docs/commands/logout.html.md.erb:
```diff
@@ -0,0 +1,34 @@
+page_title: "Commands: Logout"
+sidebar_current: "docs-commands-logout"
+description: >
+  The `logout` command provides control of both service and node logoutenance mode
```
logoutenance??

rboyer replied:
```suggestion
  The `logout` command will destroy the provided token if it was created from
`consul login`.
```

rboyer commented on website/source/docs/acl/acl-system.html.md:
```diff
@@ -27,16 +27,35 @@ At the highest level, there are two major components to the ACL system:
+   tokens. (Added in Consul 1.5)
+
+ * **ACL Service Identities** - Service Identities allow for an abbreviated
+   syntax for expressing a kind of policy needed for an token to be usable with
```
kait: This could be reworded, but I'm feeling that the likening of service identities to policies is very helpful. I think if we could flush it out here it will be really helpful for users and we could use the language in the other docs.

rboyer commented on website/source/docs/acl/acl-system.html.md:
```diff
@@ -46,6 +65,51 @@ An ACL policy is a named set of rules and is composed of the following elements:
 and will be assigned the reserved ID of `00000000-0000-0000-0000-000000000001`. This policy can be renamed but modification
 of anything else including the rule set and datacenter scoping will be prevented by Consul.
 
+### ACL Service Identities
```
kait: Can you add or link to an example?

rboyer commented on website/source/docs/acl/acl-system.html.md:
```diff
@@ -46,6 +65,51 @@ An ACL policy is a named set of rules and is composed of the following elements:
 
+### ACL Service Identities
+
+-> (Added in Consul 1.5)
```
```suggestion
-> Added in Consul 1.5.0
```

rboyer commented on website/source/docs/acl/acl-system.html.md:
```diff
@@ -27,16 +27,35 @@ At the highest level, there are two major components to the ACL system:
+   [Consul Connect](/docs/connect/index.html). At authorization time this acts
+   like an additional policy was attached, the contents of which are described
+   further below. These are directly attached to tokens and roles and are not
+   independently configured. (Added in Consul 1.5)
```
```suggestion
   independently configured. (Added in Consul 1.5.0)
```

rboyer commented on website/source/docs/acl/acl-system.html.md:
```diff
@@ -27,16 +27,35 @@ At the highest level, there are two major components to the ACL system:
+
+ * **ACL Roles** - Roles allow for grouping a set of policies and service
+   identities into a reusable higher-level entity that can be linked with many
+   tokens. (Added in Consul 1.5)
```
```suggestion
   tokens. (Added in Consul 1.5.0)
```

rboyer commented on website/source/docs/acl/acl-system.html.md:
```diff
@@ -46,6 +65,51 @@ An ACL policy is a named set of rules and is composed of the following elements:
+
+-> (Added in Consul 1.5)
+
+An ACL Service Identity is an abbreviated syntax for expressing a link to a
```
```suggestion
An ACL service identity is an abbreviated syntax for expressing a link to a
```

rboyer replied:
kait: What is the "abbreviated syntax"?

rboyer commented on website/source/docs/acl/acl-system.html.md:
````diff
@@ -46,6 +65,51 @@ An ACL policy is a named set of rules and is composed of the following elements:
+
+```hcl
+// Allow the service and its sidecar proxy to register into the catalog.
+service "{{ SERVICE_NAME_FIELD }}" {
````
```suggestion
service "<Service Name>" {
```

rboyer commented on website/source/docs/acl/acl-system.html.md:
```diff
@@ -46,6 +65,51 @@ An ACL policy is a named set of rules and is composed of the following elements:
+service "{{ SERVICE_NAME_FIELD }}" {
+	policy = "write"
+}
+service "{{ SERVICE_NAME_FIELD }}-sidecar-proxy" {
```
```suggestion
service "<Service Name>-sidecar-proxy" {
```

rboyer commented on website/source/docs/acl/acl-system.html.md:
```diff
@@ -46,6 +65,51 @@ An ACL policy is a named set of rules and is composed of the following elements:
+
+### ACL Roles
+
+-> (Added in Consul 1.5)
```
```suggestion
-> Added in Consul 1.5.0
```

rboyer commented on website/source/docs/acl/auth-methods/kubernetes.html.md:
```diff
@@ -0,0 +1,132 @@
+page_title: "Kubernetes Auth Method"
+sidebar_current: "docs-acl-auth-methods-kubernetes"
+description: |-
+  The kubernetes auth method type allows for a Kubernetes Service Account Token to be used to authenticate to Consul. This method of authentication makes it easy to introduce a Consul token into a Kubernetes Pod.
```
```suggestion
  The Kubernetes auth method type allows for a Kubernetes Service Account Token to be used to authenticate to Consul. This method of authentication makes it easy to introduce a Consul token into a Kubernetes Pod.
```

rboyer commented on website/source/docs/acl/auth-methods/kubernetes.html.md:
```diff
@@ -0,0 +1,132 @@
+---
+
+-> **1.5.0+:**  This guide only applies in Consul versions 1.5.0 and newer.
+
```
kait: Can we add a brief overview that includes who the audience is, what is covered, and link to the auth method doc? If a user finds this page without reading the auth method doc first, they will belost so linking is good.

rboyer commented on website/source/docs/acl/auth-methods/kubernetes.html.md:
```diff
@@ -0,0 +1,132 @@
+
+# Kubernetes Auth Method
+
+The `kubernetes` auth method type allows for a Kubernetes Service Account Token
```
```suggestion
The `kubernetes` auth method type allows for a Kubernetes service account token
```

rboyer commented on website/source/docs/acl/auth-methods/kubernetes.html.md:
```diff
@@ -0,0 +1,132 @@
+
+The `kubernetes` auth method type allows for a Kubernetes Service Account Token
+to be used to authenticate to Consul. This method of authentication makes it
+easy to introduce a Consul token into a Kubernetes Pod.
```
```suggestion
easy to introduce a Consul token into a Kubernetes pod.
```

rboyer commented on website/source/docs/acl/auth-methods/kubernetes.html.md:
```diff
@@ -0,0 +1,132 @@
+  client used to talk with the Kubernetes API. NOTE: Every line must end with a
+  newline (`\n`).
+
+- `ServiceAccountJWT` `(string: <required>)` - A Service Account Token (JWT)
```
kait: Capitalization? Is there a way to be a little more clear that JWT actually means JSON web token?

rboyer commented on website/source/docs/acl/auth-methods/kubernetes.html.md:
```diff
@@ -0,0 +1,132 @@
+
+## RBAC
+
+The service account corresponding to the configured
```
```suggestion
The Kubernetes service account corresponding to the configured
```

rboyer commented on website/source/docs/acl/auth-methods/kubernetes.html.md:
```diff
@@ -0,0 +1,132 @@
+## Trusted Identity Attributes
+
+The authentication step returns the following trusted identity attributes for 
+use in binding rule selectors and bind name interpolation:
```
```suggestion
use in binding rule selectors and bind name interpolation.
```

rboyer commented on website/source/docs/acl/auth-methods/kubernetes.html.md:
```diff
@@ -0,0 +1,132 @@
+Initially the
+[`ServiceAccountJWT`](/docs/acl/auth-methods/kubernetes.html#serviceaccountjwt)
+given to the Consul leader uses the TokenReview API to validate the provided
+JWT is still valid. The trusted attributes of `serviceaccount.namespace`,
```
```suggestion
JWT. The trusted attributes of `serviceaccount.namespace`,
```

rboyer commented on website/source/docs/commands/acl.html.md:
```diff
@@ -94,9 +94,12 @@ Usage: consul acl <subcommand> [options] [args]
   For more examples, ask for subcommand help or view the documentation.
 
 Subcommands:
```
kait: The capitalization in this needs to be revisited since its consistent I don't think we need to do anything immediately.

rboyer replied:
Note: this is coming from the `consul acl` subcommand output, so changing it requires updating the text embedded in the binary.

rboyer commented on website/source/docs/commands/acl/acl-auth-method.html.md.erb:
```diff
@@ -0,0 +1,259 @@
+
+Usage: `consul acl auth-method <subcommand> [options] [args]`
+
+-> **Note:** All of the examples show for the subcommands will require a valid
```
kait: See comment on acl roles command page

rboyer commented on website/source/docs/commands/acl/acl-auth-method.html.md.erb:
```diff
@@ -0,0 +1,259 @@
+`CONSUL_HTTP_TOKEN` environment variable to the tokens secret ID or pass the
+secret ID as the value of the `-token` parameter.
+
+## Common Subcommand Options
```
kait: See comment on acl roles command page

rboyer commented on website/source/docs/commands/acl/acl-auth-method.html.md.erb:
```diff
@@ -0,0 +1,259 @@
+* `-kubernetes-host=<string>` - Address of the Kubernetes API server. This flag
+  is required for `-type=kubernetes`.
+
+* `-kubernetes-service-account-jwt=<string>` - A kubernetes service account JWT
```
```suggestion
* `-kubernetes-service-account-jwt=<string>` - A Kubernetes service account JWT
```

rboyer replied:
note: check that the embedded text is ok

rboyer commented on website/source/docs/commands/acl/acl-auth-method.html.md.erb:
```diff
@@ -0,0 +1,259 @@
+* `-kubernetes-host=<string>` - Address of the Kubernetes API server. This flag
+  is required for `-type=kubernetes`.
+
+* `-kubernetes-service-account-jwt=<string>` - A kubernetes service account JWT
```
```suggestion
* `-kubernetes-service-account-jwt=<string>` - A Kubernetes service account JWT
```

rboyer replied:
note: check that the embedded text is ok

rboyer commented on website/source/docs/commands/acl/acl-binding-rule.html.md.erb:
```diff
@@ -0,0 +1,280 @@
+
+Usage: `consul acl binding-rule <subcommand> [options] [args]`
+
+-> **Note:** All of the examples show for the subcommands will require a valid
```
kait: See comment on acl roles command page

rboyer commented on website/source/docs/commands/acl/acl-binding-rule.html.md.erb:
```diff
@@ -0,0 +1,280 @@
+its ID a unique binding rule ID prefix may be specified instead of the entire
+UUID. As long as it is unique it will be resolved to the full UUID and used.
+
+## Common Subcommand Options
```
kait: See comment on acl roles command page

rboyer commented on website/source/docs/commands/acl/acl-policy.html.md.erb:
```diff
@@ -82,7 +82,7 @@ Usage: `consul acl policy create [options] [args]`
    indices should be shown for each entry.
 
-* `-name=<string>` - The new policy's name. This flag is required.
+* `-name=<string>` - The new policies name. This flag is required.
```
```suggestion
* `-name=<string>` - The new policy's name. This flag is required.
```

rboyer commented on website/source/docs/commands/acl/acl-role.html.md.erb:
```diff
@@ -0,0 +1,315 @@
+
+Usage: `consul acl role <subcommand> [options] [args]`
+
+-> **Note:** All of the examples show for the subcommands will require a valid
```
```suggestion
-> **Note:** All of the example subcommands in this document will require a valid
```

rboyer replied:
note: check pre-existing docs too

rboyer commented on website/source/docs/commands/acl/acl-role.html.md.erb:
```diff
@@ -0,0 +1,315 @@
+
+-> **Note:** All of the examples show for the subcommands will require a valid
+Consul token with the appropriate permissions.  Either set the
+`CONSUL_HTTP_TOKEN` environment variable to the tokens secret ID or pass the
```
```suggestion
`CONSUL_HTTP_TOKEN` environment variable to the token's secret ID or pass the
```

rboyer replied:
note: check pre-existing docs too

rboyer commented on website/source/docs/commands/acl/acl-role.html.md.erb:
```diff
@@ -0,0 +1,315 @@
+
+## Identifying Roles
+
+In several of the subcommands a role will have to be identified to be read,
```
kait: This sentence is a little hard to read.

rboyer commented on website/source/docs/commands/acl/acl-role.html.md.erb:
```diff
@@ -0,0 +1,315 @@
+Role updated successfully
+ID:           57147d87-6bf7-f794-1a6e-7d038c4e4ae9
+Name:         crawler
+Description:  better description
```
kait: can we add a real description forthese or just put "A description of the role"

rboyer commented on website/source/docs/commands/acl/acl-role.html.md.erb:
```diff
@@ -0,0 +1,315 @@
+Role updated successfully
+ID:           57147d87-6bf7-f794-1a6e-7d038c4e4ae9
+Name:         web-crawler
+Description:  better description
```
kait: can we add a real description forthese or just put "A description of the role"

rboyer commented on website/source/docs/commands/acl/acl-role.html.md.erb:
```diff
@@ -0,0 +1,315 @@
+$ consul acl role list
+web-crawler:
+   ID:           57147d87-6bf7-f794-1a6e-7d038c4e4ae9
+   Description:  better description
```
kait: can we add a real description forthese or just put "A description of the role"

rboyer commented on website/source/docs/commands/index.html.md:
```diff
@@ -40,6 +42,8 @@ Available commands are:
     leave          Gracefully leaves the Consul cluster and shuts down
     lock           Execute a command holding a lock
+    login          Login to Consul using an Auth Method
+    logout         Destroy a Consul Token created with Login
```
```suggestion
    logout         Destroy a Consul token created with login
```

rboyer replied:
note: update help text inside of the binary

rboyer commented on website/source/docs/commands/index.html.md:
```diff
@@ -40,6 +42,8 @@ Available commands are:
     kv             Interact with the key-value store
     leave          Gracefully leaves the Consul cluster and shuts down
     lock           Execute a command holding a lock
+    login          Login to Consul using an Auth Method
```
```suggestion
    login          Login to Consul using an auth method
```

rboyer replied:
note: update help text inside of the binary

rboyer commented on website/source/docs/acl/acl-auth-methods.html.md:
```diff
@@ -0,0 +1,127 @@
+page_title: "ACL Auth Methods"
+sidebar_current: "docs-acl-auth-methods"
+description: |-
+  An Auth Method is a component in Consul that performs authentication against a trusted external party to authorize the creation of an appropriately scoped ACL token usable within the local datacenter.
```
kait: I made some earlier comment about "Auth Method" vs "auth method". We should be consistent.

rboyer replied:
```suggestion
  An auth method is a component in Consul that performs authentication against a trusted external party to authorize the creation of an appropriately scoped ACL token usable within the local datacenter.
```

rboyer commented on website/source/docs/acl/acl-auth-methods.html.md:
```diff
@@ -0,0 +1,127 @@
+
+* **Authentication** - One or more **auth methods** should be configured with
+  details about how to authenticate application credentials. Successful
+  validation of application credentials will return a set of trusted identity
```
kait: "trusted identity attributes such as ..., ..., and ..." I think a few examples would be helpful here or we could define them somewhere.

rboyer commented on website/source/docs/acl/acl-auth-methods.html.md:
```diff
@@ -0,0 +1,127 @@
+
+# ACL Auth Methods
+
+An Auth Method is a component in Consul that performs authentication against a
```
```suggestion
An auth method is a component in Consul that performs authentication against a
```

rboyer commented on website/source/docs/acl/acl-system.html.md:
```diff
@@ -55,16 +119,19 @@ elements:
 * **Secret ID** -The bearer token used when making requests to Consul.
 * **Description** - A human readable description of the token. (Optional)
 * **Policy Set** - The list of policies that are applicable for the token.
+* **Role Set** - The list of roles that are applicable for the token. (Added in Consul 1.5)
```
```suggestion
* **Role Set** - The list of roles that are applicable for the token. (Added in Consul 1.5.0)
```

rboyer commented on website/source/docs/acl/acl-system.html.md:
```diff
@@ -55,16 +119,19 @@ elements:
 * **Description** - A human readable description of the token. (Optional)
 * **Policy Set** - The list of policies that are applicable for the token.
+* **Role Set** - The list of roles that are applicable for the token. (Added in Consul 1.5)
+* **Service Identity Set** - The list of service identities that are applicable for the token. (Added in Consul 1.5)
```
```suggestion
* **Service Identity Set** - The list of service identities that are applicable for the token. (Added in Consul 1.5.0)
```

rboyer commented on website/source/docs/acl/acl-system.html.md:
```diff
@@ -55,16 +119,19 @@ elements:
+* **Service Identity Set** - The list of service identities that are applicable for the token. (Added in Consul 1.5)
 * **Locality** - Indicates whether the token should be local to the datacenter it was created within or created in
 the primary datacenter and globally replicated.
+* **Expiration Time** - The time at which this token is revoked. (Optional; Added in Consul 1.5)
```
```suggestion
* **Expiration Time** - The time at which this token is revoked. (Optional; Added in Consul 1.5.0)
```

judithpatudith commented on website/source/api/acl/tokens.html.md:
```diff
@@ -45,15 +46,49 @@ The table below shows this endpoint's support for
+- `ExpirationTTL` `(duration: 0s)` - This is a convenience field and if set
+  will initialize the `ExpirationTime` field to a value of `CreateTime +
+  ExpirationTTL`. This field is not persisted beyond its initial use. Can be
+  specified in the form of `"10s"` or `"5m"` (i.e., 10 seconds or 5 minutes,
```
```suggestion
  specified in the form of `"60s"` or `"5m"` (i.e., 60 seconds or 5 minutes,
```

judithpatudith replied:
If the time can't be smaller than a minute the example should be a valid value.

judithpatudith commented on website/source/api/acl/tokens.html.md:
```diff
@@ -254,15 +288,47 @@ The table below shows this endpoint's support for
-   internally be resolved to the policy ID. With linking tokens internally by IDs,
-   Consul enables policy renaming without breaking tokens.
+  be applied to this token. A PolicyLink is an object with an "ID" and/or
+  "Name" field to specify a policy. With this tokens can be linked to policies
```
```suggestion
  "Name" field to specify a policy. With the PolicyLink tokens can be linked to policies
```

judithpatudith commented on website/source/api/acl/tokens.html.md:
```diff
@@ -254,15 +288,47 @@ The table below shows this endpoint's support for
+
+- `ServiceIdentities` `(array<ServiceIdentity>)` - The list of [service
+  identities](/docs/acl/acl-system.html#acl-service-identities) that should be
+  applied to the token.  Added in Consul 1.5.0.
```
```suggestion
  applied to the token. Added in Consul 1.5.0.
```

rboyer commented:
Addressing some of the feedback required updating some existing documentation for policies and tokens for consistency.

For CLI docs I just decided to split out each subcommand into its own doc (in the same style as how the `consul intention` subcommands work). This made the docs seem way more approachable and deep-linkable.

kaitlincarter-hc commented on website/source/docs/acl/acl-auth-methods.html.md:
```diff
@@ -0,0 +1,127 @@
+the local datacenter.
+
+The only supported type of auth method in Consul 1.5.0 is
+[`kubernetes`](/docs/acl/auth-methods/kubernetes.html).
```
I feel like we should capitalize Kubernetes. 
```suggestion
[`Kubernetes`](/docs/acl/auth-methods/kubernetes.html).
```

rboyer replied:
I had left it as monospace text and lowercased primarily because that's the exact string name you need to provide when configuring it as an auth method.

kaitlincarter-hc commented on website/source/docs/acl/acl-system.html.md:
```diff
@@ -46,6 +65,59 @@ An ACL policy is a named set of rules and is composed of the following elements:
+
+-> Added in Consul 1.5.0
+
+An ACL service identity is a policy template for expressing a link to a policy
```
```suggestion
An ACL service identity is an [ACL policy](/docs/acl/acl-system.html#acl-policies) template for expressing a link to a policy
```
